bindings/go/src/ucx/utils.go

```diff
@@ -24,3 +33,88 @@ func CBytes(data string) unsafe.Pointer {
 func GoBytes(p unsafe.Pointer, n uint64) []byte {
 	return C.GoBytes(p, C.int(n))
 }
+
+func family(a *net.TCPAddr) int {
+	if a == nil || len(a.IP) <= net.IPv4len {
+		return syscall.AF_INET
+	}
+	if a.IP.To4() != nil {
+		return syscall.AF_INET
+	}
+	return syscall.AF_INET6
+}
+
+// Convert GO TCP address to C
+func toSockAddr(a *net.TCPAddr) (*C.ucs_sock_addr_t, error) {
+	// We can't assing to ucs_sock_addr_t->addr reference to Go's memory,
+	// so need to allocate
+	sockaddrPtr := AllocateNativeMemory(C.sizeof_struct_sockaddr_storage)
+	sa := (*C.struct_sockaddr_storage)(sockaddrPtr)
+	var result C.ucs_sock_addr_t
+	if a == nil {
+		a = &net.TCPAddr{}
+	}
+	switch family(a) {
+	case syscall.AF_INET:
+		if len(a.IP) == 0 {
+			a.IP = net.IPv4zero
+		}
+		ip4 := a.IP.To4()
+		if ip4 == nil {
+			return nil, &net.AddrError{Err: "non-IPv4 address", Addr: a.IP.String()}
+		}
+		sa.ss_family = C.AF_INET
+		var sin *C.struct_sockaddr_in = (*C.struct_sockaddr_in)(unsafe.Pointer(sa))
+		sin.sin_port = C.htons(C.ushort(a.Port))
+		charBytes := C.CString(ip4.String())
+		C.inet_pton(C.AF_INET, charBytes, unsafe.Pointer(&sin.sin_addr.s_addr))
+		FreeNativeMemory(unsafe.Pointer(charBytes))
+	case syscall.AF_INET6:
+		// In general, an IP wildcard address, which is either
+		// "0.0.0.0" or "::", means the entire IP addressing
+		// space. For some historical reason, it is used to
+		// specify "any available address" on some operations
+		// of IP node.
+		//
+		// When the IP node supports IPv4-mapped IPv6 address,
+		// we allow a listener to listen to the wildcard
+		// address of both IP addressing spaces by specifying
+		// IPv6 wildcard address.
+		if len(a.IP) == 0 || a.IP.Equal(net.IPv4zero) {
```

Comment:
IPv6zero?

Reply:
https://github.com/golang/go/blob/d9349175ad13ab228d3224c4bff8e3e1b2936b42/src/net/ipsock_posix.go#L169
> // In general, an IP wildcard address, which is either
	// "0.0.0.0" or "::", means the entire IP addressing
	// space. For some historical reason, it is used to
	// specify "any available address" on some operations
	// of IP node.
	//
	// When the IP node supports IPv4-mapped IPv6 address,
	// we allow a listener to listen to the wildcard
	// address of both IP addressing spaces by specifying
	// IPv6 wildcard address.